gprofiler/utils/node.py

```diff
@@ -0,0 +1,223 @@
+    if (
+        debugger_url_response.status_code != 200
+        or not "application/json" in debugger_url_response.headers.get("Content-Type")
+    ):
+        raise NodeDebuggerUrlNotFound
+
+    response_json = debugger_url_response.json()
+    if (
+        not isinstance(response_json, list)
+        or len(response_json) == 0
+        or not isinstance(response_json[0], dict)
+        or not "webSocketDebuggerUrl" in response_json[0].keys()
+    ):
+        raise NodeDebuggerUrlNotFound
+
+    return response_json[0]["webSocketDebuggerUrl"]
+
+
+@retry(NodeDebuggerUnexpectedResponse, 5, 1)
+def _load_dso(sock: WebSocket, module_path: str):
+    cdp_request = {
+        "id": 1,
+        "method": "Runtime.evaluate",
+        "params": {
+            "expression": f'process.mainModule.require("{os.path.join(module_path, "linux-perf.js")}").start()',
+            "replMode": True,
+        },
+    }
+    sock.send(json.dumps(cdp_request))
+    message = sock.recv()
+    try:
+        message = json.loads(message)
+
+    except json.JSONDecodeError:
+        raise NodeDebuggerUnexpectedResponse(message)
+
+    if (
+        "result" not in message.keys()
+        or "result" not in message["result"].keys()
+        or "type" not in message["result"]["result"].keys()
+        or message["result"]["result"]["type"] != "boolean"
+    ):
+        raise NodeDebuggerUnexpectedResponse(message)
+
+
+@retry(NodeDebuggerUnexpectedResponse, 5, 1)
+def _stop_dso(sock: WebSocket, module_path: str):
+    cdp_request = {
+        "id": 1,
+        "method": "Runtime.evaluate",
+        "params": {
+            "expression": f'process.mainModule.require("{os.path.join(module_path, "linux-perf.js")}").stop()',
+            "replMode": True,
+        },
+    }
+    sock.send(json.dumps(cdp_request))
+    message = sock.recv()
+    try:
+        message = json.loads(message)
+    except json.JSONDecodeError:
+        raise NodeDebuggerUnexpectedResponse(message)
+
+    if (
+        "result" not in message.keys()
+        or "result" not in message["result"].keys()
+        or "type" not in message["result"]["result"].keys()
+        or message["result"]["result"]["type"] != "boolean"
+    ):
+        raise NodeDebuggerUnexpectedResponse(message)
+
+
+def _copy_module_into_process_ns(process: psutil.Process, musl: bool, version: str):
+    proc_root = get_proc_root_path(process)
+    libc = "musl" if musl else "glibc"
+    dest_inside_container = os.path.join("tmp", "node_module", libc, version)
+    dest = os.path.join(proc_root, dest_inside_container)
+    if os.path.exists(dest):
+        return "/" + dest_inside_container
+    src = resource_path(os.path.join("node", "module", libc, version))
+    shutil.copytree(src, dest)
+    add_permission_dir(dest, stat.S_IROTH, stat.S_IXOTH | stat.S_IROTH)
+    return "/" + dest_inside_container
+
+
+def _create_symlink_for_map(process: psutil.Process, pid_inside_ns: int):
```

Comment:
Is this required? As far as I recall, `perf` is smart enough to look for `/tmp/perf-{pid}.map` file with the NSpid, not the host PID.